addons/website_mass_mailing/static/src/js/website_mass_mailing.js

```diff
@@ -76,6 +81,15 @@ publicWidget.registry.subscribe = publicWidget.Widget.extend({
 
         subscribeBtnEl.classList.toggle('d-none', !!isSubscriber);
         thanksBtnEl.classList.toggle('d-none', !isSubscriber);
+
+        if(!isSubscriber && this._turnstile && window.top == window) {
+            const el = this._turnstile.addTurnstile('website_mass_mailing_subscribe');
+            if(el) {
+                this._turnstile.addSpinner(subscribeBtnEl);
+                el[0].classList.add('mt-3');
+                el.insertAfter(this.el);
+            }
```

Comment:
It seems to work well but I would maybe call `renderTurnstile` on these in case the pop-up appear late, it shouldn't be any more expensive. Normally it will only render automagically the first time it's inserted on the page, so if script is exceptionally fast to load the pop-up might not get a widget of its own.

Reply:
addTurnstile does not call renderTurnstile. Inserting the turnstile script in the DOM calls renderTurnstile once for every turnstile widget placeholder present in the DOM when it is run.

So the thought process was:
- some form exists in the DOM
- turnstile script is added
- turnstile widgets are initialized
- pop-up appears
-> pop-up widget is not initialized because the script was already imported

But indeed I guess pop-up snippets are present in the DOM from the start anyway so should be ok.
Still wouldn't hurt to add it but it probably won't kill me if not :man_shrugging: